src/api/hooks.ts

```diff
@@ -0,0 +1,159 @@
+import axios from "axios";
+import {
+  APIError,
+  APIGetHook,
+  APIGetHookPaginated,
+  APIPostHook,
+  RequestState,
+  Issue,
+  User,
+  UserAPIError,
+  Article,
+  Pagination,
+  APIResponse,
+  ArticleContent,
+} from "../shared/types";
+import { useState, useEffect } from "react";
+import { useAuth } from "../auth/AuthProvider";
+import { getApiUrl } from "./api";
+
+const useApiGet = <T, U extends APIError = APIError>(
```

Comment:
At some point, we'll need to add documentation. Might be a good idea to start that earlier than later.